### PR TITLE
btc-wallet: refactor and improve performance

### PR DESCRIPTION
**Pull request by tacryt-socryp**
Used to take 28s to apply the blockfilter, now takes 8s. 70% improvement.

### CODE REVIEW

File: pkg/arvo/app/btc-wallet.hoon [hunk 1]
```diff
@@ -125,73 +126,536 @@
...
-    ::
-      %btc-wallet-action
-    ?<  =(our.bowl src.bowl)
-    (handle-action:hc !<(action vase))
-    ::
-      %btc-wallet-internal
-    ?>  =(our.bowl src.bowl)
-    (handle-internal:hc !<(internal vase))
-  ==
+    ?+  mark  (on-poke:def mark vase)
```
Comment (ixv) [inline pkg/arvo/app/btc-wallet.hoon:134]: Large transfers of code between the cores make me nervous, can be hard to detect a missing `:hc`. Needs to be tested thoroughly.

File: pkg/arvo/lib/bip/b158.hoon [hunk 2]
```diff
@@ -86,39 +88,44 @@
 ::   write appends to the back
 ::
 ++  str
+  ~%  %str  ..params  ~
   |%
   ++  read-bit
+    ~/  %read-bit
```
Comment (ixv) [inline pkg/arvo/lib/bip/b158.hoon:94]: Since this change can be tested in isolation, I would be happy to approve this in its own PR, if we don't want to delay this for testing all the style changes.

File: pkg/arvo/app/btc-wallet.hoon (hunk 1 above)
Comment (tacryt-socryp) [reply pkg/arvo/app/btc-wallet.hoon:134]: Yes absolutely. For the record I've gone through and manually searched every arm call to ensure the hc is there where appropriate. I'm currently running this on my ship and it's working good. What would you like me to test?

Comment (ixv) [reply pkg/arvo/app/btc-wallet.hoon:134]: I mean, basically all of it. Delete provider, set provider again, delete wallet, set wallet again, send btc to a ship, send btc to a non-ship address, receive btc from a ship, receive btc from a non-ship address

Comment (ixv) [reply pkg/arvo/app/btc-wallet.hoon:134]: happy to show you how to set up testnet if you want to do it that way.

Comment (tacryt-socryp) [reply pkg/arvo/app/btc-wallet.hoon:134]: Yeah, can we spin up a testnet tomorrow and go through it?

Comment (ixv) [reply pkg/arvo/app/btc-wallet.hoon:134]: Sure

**Review by ixv — APPROVED** [review]

**Comment by ixv** [comment]
We tested it all and it looks good
